apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterQueueSpec defines the desired state of ClusterQueue
+type ClusterQueueSpec struct {
+	// resourceGroups describes groups of resources.
+	// Each resource group defines the list of resources and a list of flavors
+	// that provide quotas for these resources.
+	// resourceGroups can be up to 16.
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=16
+	ResourceGroups []ResourceGroup `json:"resourceGroups,omitempty"`
+
+	// cohort that this ClusterQueue belongs to. CQs that belong to the
+	// same cohort can borrow unused resources from each other.
+	//
+	// A CQ can be a member of a single borrowing cohort. A workload submitted
+	// to a queue referencing this CQ can borrow quotas from any CQ in the
+	// cohort. Only quotas listed in the CQ can be borrowed.
+	// If empty, this ClusterQueue cannot borrow from any other ClusterQueue and
+	// vice versa.
+	//
+	// Validation of a cohort name is equivalent to label keys. These are just
+	// names to link CQs together, and they are meaningless otherwise.
+	Cohort string `json:"cohort,omitempty"`
+
+	// QueueingStrategy indicates the queueing strategy of the workloads
+	// across the queues in this ClusterQueue. This field is immutable.
+	// Current Supported Strategies:
+	//
+	// - StrictFIFO: workloads are ordered strictly by creation time.
+	// Older workloads that can't be admitted will block admitting newer
+	// workloads even if they fit available quota.
+	// - BestEffortFIFO：workloads are ordered by creation time,
```

Comment:
```suggestion
	// - BestEffortFIFO: workloads are ordered by creation time,
```

Reply:
hahah weird unicode character?
How did you spot that?!

Reply:
In Japanese (my native language), we always use double-byte characters by default. So I easily find the double-byte characters :)